### ISSUE TITLE
Comments with string embedded

### ISSUE THREAD

**Issue by Chris00** [ocaml/tuareg#95]
The following
```
(* " *)
comment
(* " *)
```
is a single comment containing a string.  The syntax highlighting must be updated for that.

**Comment by Chris00** [comment]
Putting the cursor on _comment_ and running `(nth 4 (syntax-ppss))` return `nil`.

**Comment by monnier** [comment]
Yes, this is a nasty one to handle.  I guess one way to do that is to add to syntax-propertize-function a check that looks at every `*)`: if we're inside a comment before the "*", then call `parse-partial-sexp` from the beginning of the comment (right after the `(*`) and check `(nth 3 pps)`.

**Comment by cfcs** [comment]
I just got bitten by this :(
https://github.com/ocaml/merlin/issues/824

**Comment by Chris00** [comment]
See alse https://github.com/ocaml/ocaml/pull/1901#issuecomment-404818182
